### ISSUE TITLE
neuronx compiler unusual behaviors on ConvBERT / XLM / FlauBERT

### ISSUE THREAD

**Issue by JingyaHuang** [huggingface/optimum-neuron#18]
* __Issue 1__: ConvBERT  - tracing failed on INF2

<details>
<summary>Error log</summary>
<br>

```
03/01/2023 01:28:13 PM ERROR 59059 [Tensorizer]: Transformation error on operator: _dot.1235
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]: ***************************************************************
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:  An Internal Compiler Error has occurred
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]: ***************************************************************
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]: 
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]: Error message:  13824i_0+768i_1+k
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]: 
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]: Error class:    AssertionError
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]: Error location: Unknown
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]: Command line:   /home/ubuntu/aws_neuron_venv_pytorch/bin/neuronx-cc compile /tmp/tmptflduq47/model --framework XLA --target trn1 --output /tmp/tmptflduq47/graph.neff
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]: 
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]: Internal details:
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/driver/CommandDriver.py", line 235, in neuronxcc.driver.CommandDriver.CommandDriver.run
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/driver/commands/CompileCommand.py", line 1014, in neuronxcc.driver.commands.CompileCommand.CompileCommand.run
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/driver/commands/CompileCommand.py", line 965, in neuronxcc.driver.commands.CompileCommand.CompileCommand.runPipeline
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/driver/commands/CompileCommand.py", line 990, in neuronxcc.driver.commands.CompileCommand.CompileCommand.runPipeline
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/driver/commands/CompileCommand.py", line 994, in neuronxcc.driver.commands.CompileCommand.CompileCommand.runPipeline
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/driver/Job.py", line 300, in neuronxcc.driver.Job.SingleInputJob.run
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/driver/Job.py", line 326, in neuronxcc.driver.Job.SingleInputJob.runOnState
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/driver/Pipeline.py", line 30, in neuronxcc.driver.Pipeline.Pipeline.runSingleInput
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/driver/Job.py", line 300, in neuronxcc.driver.Job.SingleInputJob.run
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/driver/Job.py", line 326, in neuronxcc.driver.Job.SingleInputJob.runOnState
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/driver/jobs/Frontend.py", line 591, in neuronxcc.driver.jobs.Frontend.Frontend.runSingleInput
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/driver/jobs/Frontend.py", line 387, in neuronxcc.driver.jobs.Frontend.Frontend.runXLAFrontend
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/Frontend.py", line 168, in neuronxcc.starfish.penguin.Frontend.tensorizeXla
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/Frontend.py", line 243, in neuronxcc.starfish.penguin.Frontend.tensorizeXlaImpl
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/Frontend.py", line 244, in neuronxcc.starfish.penguin.Frontend.tensorizeXlaImpl
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/Frontend.py", line 266, in neuronxcc.starfish.penguin.Frontend.tensorizeXlaImpl
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/Compile.py", line 183, in neuronxcc.starfish.penguin.Compile.compile_cu
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/Compile.py", line 185, in neuronxcc.starfish.penguin.Compile.compile_cu
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/Compile.py", line 195, in neuronxcc.starfish.penguin.Compile.compile_cu
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/DotTransform.py", line 432, in neuronxcc.starfish.penguin.DotTransform.PassManager.transformFunction
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/DotTransform.py", line 442, in neuronxcc.starfish.penguin.DotTransform.PassManager.transformFunction
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/DotTransform.py", line 152, in neuronxcc.starfish.penguin.DotTransform.DotTransform.run_or_rollback
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/DotTransform.py", line 196, in neuronxcc.starfish.penguin.DotTransform.DotTransform.run_with_exception_handling
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/DotTransform.py", line 178, in neuronxcc.starfish.penguin.DotTransform.DotTransform.run_with_exception_handling
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/DotTransform.py", line 208, in neuronxcc.starfish.penguin.DotTransform.DotTransform.timed_run_
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/DotTransform.py", line 210, in neuronxcc.starfish.penguin.DotTransform.DotTransform.timed_run_
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/DotTransform.py", line 211, in neuronxcc.starfish.penguin.DotTransform.DotTransform.timed_run_
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/DotTransform.py", line 381, in neuronxcc.starfish.penguin.DotTransform.IterativeTransform.run_
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/DotTransform.py", line 382, in neuronxcc.starfish.penguin.DotTransform.IterativeTransform.run_
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/DotTransform.py", line 396, in neuronxcc.starfish.penguin.DotTransform.IterativeTransform.iterate
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/DotTransform.py", line 334, in neuronxcc.starfish.penguin.DotTransform.DotTransform.transformFunction
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/DotTransform.py", line 329, in neuronxcc.starfish.penguin.DotTransform.DotTransform.runTransforms
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/DotTransform.py", line 318, in neuronxcc.starfish.penguin.DotTransform.DotTransform.transformStmts
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/DotTransform.py", line 133, in neuronxcc.starfish.penguin.DotTransform.DotTransform.transform
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/DotTransform.py", line 366, in neuronxcc.starfish.penguin.DotTransform.DotTransform.transformBasicBlock
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/DotTransform.py", line 369, in neuronxcc.starfish.penguin.DotTransform.DotTransform.transformBasicBlock
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/DotTransform.py", line 133, in neuronxcc.starfish.penguin.DotTransform.DotTransform.transform
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/DotTransform.py", line 356, in neuronxcc.starfish.penguin.DotTransform.DotTransform.transformStmt
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/DotTransform.py", line 133, in neuronxcc.starfish.penguin.DotTransform.DotTransform.transform
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/DotTransform.py", line 356, in neuronxcc.starfish.penguin.DotTransform.DotTransform.transformStmt
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/DotTransform.py", line 133, in neuronxcc.starfish.penguin.DotTransform.DotTransform.transform
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/DotTransform.py", line 356, in neuronxcc.starfish.penguin.DotTransform.DotTransform.transformStmt
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/DotTransform.py", line 133, in neuronxcc.starfish.penguin.DotTransform.DotTransform.transform
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/DotTransform.py", line 356, in neuronxcc.starfish.penguin.DotTransform.DotTransform.transformStmt
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/DotTransform.py", line 133, in neuronxcc.starfish.penguin.DotTransform.DotTransform.transform
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/targets/tonga/passes/CommuteConcat.py", line 770, in neuronxcc.starfish.penguin.targets.tonga.passes.CommuteConcat.CommuteConcat.transformTensorContractOp
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/targets/tonga/passes/CommuteConcat.py", line 703, in neuronxcc.starfish.penguin.targets.tonga.passes.CommuteConcat.should_rewrite_access_optimize_tc_operand
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/targets/tonga/passes/CommuteConcat.py", line 446, in neuronxcc.starfish.penguin.targets.tonga.passes.CommuteConcat.extract_weight_contract_dim
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   File "neuronxcc/starfish/penguin/ir/AffineExpr.py", line 74, in neuronxcc.starfish.penguin.ir.AffineExpr.AsTrivialExpr
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]: 
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]: Version information:
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   NeuronX Compiler version 2.4.0.21+b7621be18
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   HWM version 2.4.0.1-90172456c
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   NEFF version Dynamic
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   TVM not available
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   NumPy version 1.20.3
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]:   MXNet not available
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]: 
03/01/2023 01:28:13 PM ERROR 59059 [neuronx-cc]: Artifacts stored in: /home/ubuntu/optimum-neuron/neuronxcc-n8epgszg
```

</details>

* __Issue 2__ : XLM and flauBERT have higher than usual(`1e-5`) difference on INF2 compared to PyTorch on CPU
    * xlm
   
  ```
  E   AssertionError: xlm, sequence-classification -> The maximum absolute difference between the output of the reference model and the Neuron exported model is not within the set tolerance 0.0001:
  E   - logits: max diff = 0.09034809470176697
  ```

    * Flaubert
 
  ```
  E   AssertionError: flaubert, sequence-classification -> The maximum absolute difference between the output of the reference model and the Neuron exported model is not within the set tolerance 0.0001:
  E   - logits: max diff = 0.04413249343633652
  ```

**Comment by JingyaHuang** [comment]
__[Reproduction]__

```python
import copy
from pathlib import Path

from transformers import AutoModelForSequenceClassification

from optimum.neuron.exporter import export, validate_model_outputs
from optimum.neuron.exporter.model_configs import BertNeuronConfig, DistilBertNeuronConfig

model_id = "YituTech/conv-bert-base"
model = AutoModelForSequenceClassification.from_pretrained(model_id)
reference_model = copy.deepcopy(model)
neuron_config = BertNeuronConfig(config=model.config, task="sequence-classification", batch_size=2, sequence_length=18)
output_path = Path("model.pt")

export(
    model=model,
    config=neuron_config,
    output=output_path,
)

# Validation
run = 10
value_failures = []
for _ in range(run):
    validate_model_outputs(
        config=neuron_config,
        reference_model=reference_model,
        neuron_model_path=output_path,
        neuron_named_outputs=["logits"],
    )

```

**Comment by JingyaHuang** [comment]
Fixed by #33
